File: backend/src/main/java/com/quickcart/auth/service/AuthService.java
package com.quickcart.auth.service;

import com.quickcart.auth.dto.AuthResponse;
import com.quickcart.auth.dto.LoginRequest;
import com.quickcart.auth.dto.RegisterRequest;

public interface AuthService {
	AuthResponse register(RegisterRequest request);
	AuthResponse login(LoginRequest request);
}

File: frontend/src/services/authService.js
import apiClient from '../lib/apiClient'

export async function loginRequest(payload) {
  const { data } = await apiClient.post('/auth/login', payload, {
    headers: { 'Content-Type': 'application/json' },
  })
  return data
}

export async function registerRequest(payload) {
  const { data } = await apiClient.post('/auth/register', payload, {
    headers: { 'Content-Type': 'application/json' },
  })
  return data
}

export async function getCurrentUserRequest() {
  const { data } = await apiClient.get('/user/me')
  return data
}

export async function logoutRequest() {
  const { data } = await apiClient.post('/auth/logout')
  return data
}
